org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java

```diff
@@ -179,7 +184,18 @@ private void resolveRequestedBindingKeys(JavacBindingResolver bindingResolver, S
 			cu.accept(new BindingBuilder(bindingMap));
 		}
 
-		NameEnvironmentWithProgress environment = new NameEnvironmentWithProgress(cp, null, monitor);
+		INameEnvironment environment = null;
+		if (project != null) {
```

Comment:
you can instead try `if (project instanceof JavaProject internalProject)` to reuse internalProject later instead of casting.